dist/ruff_check.py

```diff
@@ -1,48 +1,83 @@
+        print(e)
+        return False
+
+    if not output == f"ruff {expected_version}":
+        print(
+            f"Unexpected Ruff version detected: {output} instead of {expected_version}"
+        )
+        return False
+    return True
+
+
+def get_expected_ruff_version(config_file: str) -> str:
+    ruff_version = None
+    with open(config_file) as file:
+        for line in file:
+            m = re.search(r'required-version = "==(\d.\d.\d)"', line.strip())
+            if m:
+                ruff_version = m.group(1)
+                break
+    assert ruff_version, f"No ruff version has been found in {RUFF_TOML_FILE}"
+    return ruff_version
 
 
-def run(cmd):
+def run_ruff_checks(cmd: list[str]) -> bool:
     try:
-        output = subprocess.check_output(cmd, cwd=current_dir).decode().strip()
+        output = subprocess.check_output(cmd, cwd=CURRENT_DIR).decode().strip()
         if output == "All checks passed!":
             return True
         else:
             print(output)
             return False
     except subprocess.CalledProcessError as cpe:
-        print("The command [%s] failed:\n%s" % (" ".join(cmd), cpe.output.decode("utf-8")))
+        print(
+            "The command [%s] failed:\n%s" % (" ".join(cmd), cpe.output.decode("utf-8"))
+        )
         return False
 
 
-if not which("ruff"):
+def show_ruff_installation_steps(expected_version: str) -> None:
     doc_link = "https://docs.astral.sh/ruff/installation/"
-    lines = []
-    ruff_version = None
-    with open(ruff_config) as file:
-        for line in file:
-            m = re.search(r'required-version = "==(\d.\d.\d)"', line.strip())
-            if m:
-                ruff_version = m.group(1)
-                break
-    print("Ruff is not installed!")
     print(f"Please install Ruff using the official documentation: {doc_link}\n")
     print("Suggested steps using a virtual environment:")
-    print(f"virtualenv -p python3 {current_dir.parent}/venv")
-    print(f"source {current_dir.parent}/venv/bin/activate")
-    print(f"python3 -m pip install ruff=={ruff_version}\n")
+    print(f"virtualenv -p python3 {CURRENT_DIR.parent}/venv")
+    print(f"source {CURRENT_DIR.parent}/venv/bin/activate")
+    print(f"python3 -m pip install ruff=={expected_version}\n")
+
+
+ruff_expected_version = get_expected_ruff_version(RUFF_TOML_FILE)
+
+# If Ruff is not installed, show how to install it.
+if not which("ruff"):
+    print("Ruff is not installed!")
+    show_ruff_installation_steps(ruff_expected_version)
+    sys.exit(1)
+# Otherwise, check which version is installed.
+elif not check_ruff_version(ruff_expected_version):
+    print("Ruff version check failed.")
+    show_ruff_installation_steps(ruff_expected_version)
     sys.exit(1)
 
-cmd = ["ruff", "check", "--fix", "../.", ".", "--config", ruff_config]
 
-if not run(cmd):
+cmd = ["ruff", "check", "--fix", "../.", ".", "--config", RUFF_TOML_FILE]
+
+if not run_ruff_checks(cmd):
     sys.exit(1)
```

Comment:
How about enclosing these lines of code in a `if __name__ == '__main__':` block to make it clear they will be executed when the script is called without argument (by s_all)?

Reply:
Sounds good - see a9ed3400c1696558c6be7821ed8ee690dca23d3f

Reply:
LGTM!